Classes/Base/Context.php

```diff
@@ -330,4 +331,49 @@ protected static function addGroups( \Aimeos\MShop\Context\Item\Iface $context )
 
 		return $context;
 	}
+
+
+	/**
+	 * Adds the frontend date time to the context
+	 *
+	 * @param \Aimeos\MShop\Context\Item\Iface $context Context object including config
+	 * @return \Aimeos\MShop\Context\Item\Iface Modified context object
+	 */
+	protected static function addDateTime( \Aimeos\MShop\Context\Item\Iface $context )
+	{
+		// Handle the admin panel, according to the version number.
+		if ( version_compare( TYPO3_version, '9.2.0', '<' )
+			&& isset($GLOBALS['TSFE'])
+		) {
+			// The old admin panel saves it's stuff inside the user settings of
+			// the current admin user. These settings will get used by the core,
+			// even if the actual panel is deactivated.
+			// To have a similar behavior, we also need to look at these settings.
+			if ( isset($GLOBALS['BE_USER']->uc['TSFE_adminConfig']['display_top'] )
+				&& isset($GLOBALS['BE_USER']->uc['TSFE_adminConfig']['preview_simulateDate'] )
+				&& $GLOBALS['BE_USER']->uc['TSFE_adminConfig']['display_top'] === '1'
+				&& !empty( (int)$GLOBALS['BE_USER']->uc['TSFE_adminConfig']['preview_simulateDate'] ) )
+			{
+				return $context->setDateTime(
+					date( 'Y-m-d H:i:s', (int) $GLOBALS['BE_USER']->uc['TSFE_adminConfig']['preview_simulateDate'] )
+				);
+			}
+		}
+		elseif ( isset( $GLOBALS['BE_USER']->adminPanel ) )
+		{
+			// Read the actual configuration and then use it.
+			$time = GeneralUtility::makeInstance( \TYPO3\CMS\Adminpanel\Service\ConfigurationService::class )
```

Comment:
What is the exact content of `$time`?

Reply:
`2019-08-01T22:00:00.000Z` in the new admin panel.

A timetamp in the old one.

Reply:
I've experimented a little bit with the $time parameter. It is possible to "inject" crap here, with the help of the browser inspector. Nothing security related, but it is possible to set it to 'qwer'. The core then simply ignores the invalid data. We should do the same here.

Reply:
Then, you should use `empty()` on the output of `strtotime()` to catch "0" and "false"

Reply:
Done.